### PR TITLE
Help info as tooltips for forms

### PR DESCRIPTION
**Pull request by christophscheuing**
see issue: #705

### Visible/Frontend Changes
yes

### Architectural/Backend Changes
no

### CODE REVIEW

**Comment by christophscheuing** [comment]
Does allowing such tooltips make sense to you?
I'm note sure how to set the [matTooltipDisabled]="!editing" via the config file.
Maybe we should also add a possibility for a delay for the tooltip to appear?

**Comment by TheSlimvReal** [comment]
I guess providing the option for definition tooltips through the config makes sense. However it requires quite some text in the config if a tooltip for each field is provided and the delay also makes sense. 
Using `[matTooltipDisabled]="!editing"` should already be working, this should require anything special in the config file.

**Comment by sleidig** [comment]
I also think tooltips can be an important addition and should be set in the config.

An alternative approach for the UI could be to show a (?) icon next to the component and only add the tooltip to that. That would make the help text more explicitly noticeable (especially on mobile) but might also clutter the interface a lot. Not sure what works best for us or what best practices are regarding this.

File: src/app/core/entity-components/entity-details/form/form.component.html [hunk 1]
```diff
@@ -112,7 +112,10 @@
           </mat-form-field>
         </div>
 
-        <div *ngIf="row.input === 'select'">
+        <div *ngIf="row.input === 'select'"
+          [matTooltip]="row.tooltip"
+          [matTooltipPosition]="row.tooltipPosition"
```
Comment (sleidig) [inline src/app/core/entity-components/entity-details/form/form.component.html:117]: couldn't you add this to the parent div (row 9) and automatically make it available for all types?

Comment (christophscheuing) [reply src/app/core/entity-components/entity-details/form/form.component.html]: Yes, that would work; thank you for that input.

**Comment by christophscheuing** [comment]
I now moved the tooltips to a question mark icon shown at the end of the field when editing. I don't know how you like this UI.

**Comment by christophscheuing** [comment]
It also works (more or less) on mobile. On my iPad a simple tap on the question mark icon shows the tooltip. On the iPhone I have to do a long press on the icon.

What might be a little inconvenient is the fact that he tooltip is sometimes hidden behind to entries of the drop down menu on mobile. I don't know if there's a good solution to that.

What do you think?

File: src/app/core/entity-components/entity-details/form/form.component.scss [hunk 2]
```diff
@@ -46,3 +46,8 @@
   padding-left: 10px;
   color: black;
 }
+
+.mat-icon{
+  font-size: 1em;
```
Comment (sleidig) [inline src/app/core/entity-components/entity-details/form/form.component.scss:51]:
```suggestion
  font-size: 1.25em;
  color: grey;
```
let's make them a bit bigger to make it an easier target especially on mobile

**Comment by sleidig** [comment]
Looks great to me!
I like the style with the question mark icon, this clearly shows the user that there is extra help available if needed.
Mostly works well on mobile as well. I don't think it's much of a problem that it is covered by the opened dropdown. Only sometimes I cannot reliably touch the icon in active fields, focusing the form field rather than displaying the tooltip.

Didn't you want to move up the help icon rather then duplicating it for every type?

**Review by sleidig — APPROVED** [review]

File: src/app/core/config/config-fix.ts [hunk 3]
```diff
@@ -719,11 +719,13 @@ export const defaultConfig = {
                   [
                     {
                       "input": "age",
+                      "tooltip": "This field is read-only. Edit Date of Birth to change age. Select Jan 1st if you only know the year of birth.",
                       "id": "dateOfBirth",
                       "placeholder": "Date of Birth"
                     },
                     {
                       "input": "select",
+                      "tooltip": "This is a sample toopltip for a select field. Please select the biological sex of the person.",
```
Comment (sleidig) [inline src/app/core/config/config-fix.ts:728]:
```suggestion
```
cleanup before merge

File: src/app/core/config/config-fix.ts [hunk 4]
```diff
@@ -733,6 +735,7 @@ export const defaultConfig = {
                     },
                     {
                       "input": "text",
+                      "tooltip": "This is a sample toopltip for a textfield. Please enter the native language.",
```
Comment (sleidig) [inline src/app/core/config/config-fix.ts:738]:
```suggestion
```
cleanup before merge

File: src/app/core/config/config-fix.ts [hunk 5]
```diff
@@ -750,6 +753,7 @@ export const defaultConfig = {
                     },
                     {
                       "input": "configurable-enum-select",
+                      "tooltip": "This is a sample tooltip for a configurable-enum-select-field. Please choose Adahar status.",
```
Comment (sleidig) [inline src/app/core/config/config-fix.ts:756]:
```suggestion
```
cleanup before merge

**Pull request merged**